packages/styled-components/src/utils/hash.js

```diff
@@ -1,21 +1,22 @@
 // @flow
 /* eslint-disable */
 
-export const SEED = 5381 | 0;
+export const SEED = 5381;
 
 // When we have separate strings it's useful to run a progressive
 // version of djb2 where we pretend that we're still looping over
 // the same string
 export const phash = (h: number, x: string): number => {
-  h |= 0;
-  for (let i = 0, l = x.length | 0; i < l; i++) {
-    h = (h << 5) + h + x.charCodeAt(i);
+  let i = x.length;
+
+  while (i) {
```

Comment:
can we work this forwards possibly? That should in theory keep some of the hashes the same as before?